src/tests/unit/tests/background/permissions-state-tracker.test.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Interpreter } from 'background/interpreter';
+import { PermissionsStateTracker } from 'background/permissions-state-tracker';
+import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
+import { Message } from 'common/message';
+import { Messages } from 'common/messages';
+import { tick } from 'tests/unit/common/tick';
+import { IMock, It, Mock, Times } from 'typemoq';
+
+describe('PermissionsStateTracker', () => {
+    let testSubject: PermissionsStateTracker;
+    let browserAdapterMock: IMock<BrowserAdapter>;
+    let interpreterMock: IMock<Interpreter>;
+
+    beforeEach(() => {
+        browserAdapterMock = Mock.ofType<BrowserAdapter>();
+        interpreterMock = Mock.ofType<Interpreter>();
+
+        testSubject = new PermissionsStateTracker(browserAdapterMock.object, interpreterMock.object);
+    });
+
+    it.each([true, false])('notifyChange sends the expected interpreter message', async permissionsState => {
+        browserAdapterMock
+            .setup(adapter => adapter.containsPermissions(It.isAny()))
```

Comment:
Probably this should setup should require that it gets the expected origin pattern, rather than accepting any input